app.rb

```diff
@@ -779,11 +779,15 @@ def render_template(name)
       error 400,
         :error => _("The nic_max parameter must be an integer not starting with 0")
 
-    params['http_port'].nil? and request.secure? and
-      error 400,
-        :error => _("The `http_port` argument must be supplied for bootstrap generation on a secure port")
-
-    @bootstrap_port = params['http_port'].nil? ? request.port.to_s : params['http_port']
+    params["http_port"].nil? or (params["http_port"] =~ /\A[1-9][0-9]?[0-9]?[0-9]?[0-9]?\Z/ and
+      params['http_port'].to_i < 65536) or error 400,
+        :error => _('The http_port parameter must be an integer between 1 and 65535')
+
+    @bootstrap_port = params['http_port']
+    @bootstrap_port ||= ENV['HTTP_PORT']
```

Comment:
This will be reached if the `http_port` parameter is not supplied. The `params["http_port"].nil?` short-circuits the rest of the part where we're checking that the parameter is an integer.